cmake/Modules/FindBLAS.cmake

```diff
@@ -331,6 +331,16 @@ int main() {
   SET(CMAKE_REQUIRED_LIBRARIES)
 ENDIF(BLAS_LIBRARIES)
 
+# Blas has bfloat16 support?
+IF(BLAS_LIBRARIES)
+  SET(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
+  check_function_exists("sbgemm_" BLAS_HAS_SBGEMM)
+  set(CMAKE_REQUIRED_LIBRARIES)
+  IF(BLAS_HAS_SBGEMM)
+    add_compile_options(-DBLAS_HAS_SBGEMM)
+  ENDIF(BLAS_HAS_SBGEMM)
```

Comment:
Actually, this condition does not belong in FindBLAS.cmake, as module should not make any changes to project compiler flags.
A better solution would be to move this logic to `Dependencies.cmake` and perhaps preserve it in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Config.h.in rather than pass as compiler option.

Reply:
@malfet 
I got some comments from a very friendly and helpful colleague who is trying to help me understand what you are suggesting.  Here are his comments:

=========================================================
Looks like he’s just saying that the BLAS-module-specific code here responsible for finding whether BLAS is present and describing it, doesn’t really have any business controlling how the broader project builds (maybe the broader project wants to inquire about whether several different BLASes are installed, and then choose from among them or something).

So the better solution would be to have `FindBLAS` set the `BLAS_HAS_SBGEMM` cmake flag (as it already does), but avoid setting the compile option.

Instead, I guess he’s suggesting something like:
- update the `elseif(BLAS STREQUAL "Generic")` block in `cmake/Dependencies.cmake` to test for `BLAS_HAS_SBGEMM` and `set()` some other appropriate variable (`AT_BUILD_WITH_BLAS_SBGEMM`?)
- add a line in `aten/src/ATen/Config.h.in` to turn that into a function-like macro: `#define AT_BUILD_WITH_BLAS_SBGEMM() @AT_BUILD_WITH_BLAS_SBGEMM@`
- and then modify `.../CPUBlas.cpp` to use that new function-like macro instead of `BLAS_HAS_SBGEMM`, e.g.: `#if AT_BUILD_WITH_BLAS() && AT_BUILD_WITH_BLAS_SBGEMM()`
=========================================================

We both agree it's best to run this by you, so you can get us back on track if we failed to undetstand your wishes, before we go chopping up the code any more...  please understand I am happy to do what it takes, but it makes it easier for me if you give me specific steps, as I'm an algorithmicist with **inadequate** skills at fancy makefile stuff like this!